modxcms/revolution
# [3.x] Restrict tag usage and ensure proper escaping of element name, caption, and description fields

### What does it do?

Strips tags and html encodes the name, caption/label, and description fields for elements—both in the back end (processors) and front end (js forms, grids, and windows). Made separate commits for processors and js in case that makes testing easier.

### Why is it needed?

There should be more appropriate limitations on HTML tag usage within certain fields. The initial submission of this PR is highly restrictive, which I realize may need to be loosened depending on reviewer feedback.  

### How to test
 
1. Note, There are three areas covered that should be tested:
    - The main element create/edit form pages
    - The quick create/edit windows
    - The form customization grid (you'll need to create a new profile and set to test)
2. From your terminal app, run `grunt build` from within the `_build/templates/default` folder and clear your browser cache.
2. Cherry pick just the processors commit to verify tags can not be saved to the database for the _name/templatename_, and only select tags and attributes can be saved for _caption_ (TVs only) and _description_ fields for all elements (set with the 4 new system settings `elements_caption_allowedtags`, `elements_caption_allowedattr`, `elements_description_allowedtags`, and `elements_description_allowedattr`).
3. Add in the javascript commit and verify that tags are prevented in these form fields and special characters are html encoded as well.

### Related issue(s)/PR(s)
Resolves #15925
